src/ExpressionEvaluator/CSharp/Test/ResultProvider/ValueFormattingTests.cs

```diff
@@ -214,8 +214,8 @@ public void String()
             Assert.Equal("\"\\uffff\"", FormatValue(s, useHexadecimal: true));
 
             string multiByte = "\ud83c\udfc8";
-            Assert.Equal(string.Format(format, "🏈"), FormatValue(multiByte));
-            Assert.Equal(string.Format(format, "🏈"), FormatValue(multiByte, useHexadecimal: true));
+            Assert.Equal(string.Format(format, "\\ud83c\\udfc8"), FormatValue(multiByte));
+            Assert.Equal(string.Format(format, "\\ud83c\\udfc8"), FormatValue(multiByte, useHexadecimal: true));
```

Comment:
@gafter, I didn't think the language  required input to be UTF-8? Only that a conforming compiler had to support at-least UTF-8.

I would think, if the compiler supports UTF-16 and the user decides to use UTF-16 characters, then the compiler should just work (and should not care if the sequence can be represented in UTF-8 or not).

The spec currently (and as far back as I can check) defines the various characters as:
```
letter-character:
  A Unicode character of classes Lu, Ll, Lt, Lm, Lo, or Nl 
  A unicode-escape-sequence representing a character of classes Lu, Ll, Lt, Lm, Lo, or Nl
combining-character:
  A Unicode character of classes Mn or Mc 
  A unicode-escape-sequence representing a character of classes Mn or Mc
decimal-digit-character:
  A Unicode character of the class Nd 
  A unicode-escape-sequence representing a character of the class Nd
connecting-character:  
  A Unicode character of the class Pc
  A unicode-escape-sequence representing a character of the class Pc
formatting-character:  
  A Unicode character of the class Cf
  A unicode-escape-sequence representing a character of the class Cf
```